### ISSUE TITLE
Provide more ways to identify PVs in the custom pricing CSV file

### ISSUE THREAD

**Issue by kirbsauce** [kubecost/cost-analyzer-helm-chart#1018]
Currently, metadata.name, metadata.labels, and metadata.annotations are the supported ways to match the PV.  Could other methods be supported, such as "spec.ProviderId"?

gz#653

┆Issue is synchronized with this [Jira Task](https://kubecost.atlassian.net/browse/GIB-34) by [Unito](https://www.unito.io)

**Comment by danielnaguri** [comment]
Thanks @kirbsauce. Adding to this we would also love for cost model to automatically determine the costs for Missing PV in Billing data based on the class match (Type of PV).

**Comment by kirbsauce** [comment]
thanks @danielnaguri!  My understanding is this occurs during the 2-3 day lag of new PVs showing up in the billing data, correct?

**Comment by danielnaguri** [comment]
Yes, Exactly !

**Comment by kirbsauce** [comment]
One example:

From AWS billing, it can see that the identifiers for pv are of the form vol-xxxxxxxxxxxxxxxxx.

Initial research places the identifier in an arbitrary PV field that depends on the kind of disk provisioner.  This means it will take a little bit longer than adding support for a higher-level, provisioner-agnostic identifier, but certainly doable.

**Comment by AjayTripathy** [comment]
It looks like our reflection on arbitrary fields in the spec for PVs is not complete and there's complexity here because where the identifier exists in k8s is dependent on the disk provisioner. I'd expect an ETA here for v1.86.0 in 2 weeks or so.

**Comment by kirbsauce** [comment]
@AjayTripathy , unless I misunderstand the proximity to this particular issue, my thinking is that "automatically determining costs for Missing PVs" is something we should track in a separate issue.  Do you agree?

**Comment by AjayTripathy** [comment]
If we're talking about using an estimate for the custom pricing provider when the CSV does not provide a value, I think it's fine to track here.

**Comment by kirbsauce** [comment]
That's what we're talking about.  Will track in parallel.

**Comment by kirbsauce** [comment]
@AjayTripathy , a wish list of provisioners to support...

AWS EBS General Purpose SDD
AWS EBS General Purpose SDD with increased iops & throughput
AWS EBS Provisioned IOPS SDD
AWS EBS Cold HDD
AWS EBS Throughput Optimized HDD

Azure Files Standard
Azure Files Premium
Azure Premium SDD
Azure Ultra SDD
